### DISCUSSION TITLE
trained network in tf to pytorch

### DISCUSSION THREAD

**Discussion by Sma6500** [voxelmorph/voxelmorph#400] [Q&A] [answered]
Hi, 
for transfert learning purpose I would like to transform the given net : vxm_dense_brain_T1_3D_mse.h5 from tensorflow to pytorch.
I tried to use the MMdNN library from microsoft : 
mmconvert -sf keras -iw vxm_dense_brain_T1_3D_mse.h5 -df pytorch -om keras_to_torch.pt
But it raised me the following error : 
ValueError: Unknown layer: VxmDense. Please ensure this object is passed to the `custom_objects` argument. 
And I am not sure of how I could solve it. I don't even know if it's possible given the fact that Voxelmorph Dense net is a specific net with specific block. 
Do you have any advice to get the trained weights into a pytorch format ?

Thanks you !! :)

**Comment by adalca** [comment, marked as answer]
I don't have experience with this, but you could get the weights via tensorflow (model.get_weights()), which is just a series of lists and numpy arrays. You could then load this into your pytorch.

@ahoopes or @balakg might know more?

**Reply by ahoopes** [reply]
One thing to consider is that the pytorch models use a channels-first architecture while the TF models use channels-last, so you might need to move the last axis of the kernel weight arrays.
